### ISSUE TITLE
Enhancement: Extend Context Parameters in Test Function to Match Train Function

### ISSUE THREAD

**Issue by AlfredQin** [Deci-AI/super-gradients#1331]
### Description

In the `sg_trainer.py` file, there appears to be a discrepancy between the context parameters used in the `train` and `test` functions. The `context` in the `test` function has fewer parameters compared to the `context` in the `train` function.

### Details

In the `train` function, `context` is instantiated with multiple parameters, including `optimizer`, `net`, `experiment_name`, `ckpt_dir`, `criterion`, `lr_warmup_epochs`, and others. Here is the related code:

```python
context = PhaseContext(
    optimizer=self.optimizer,
    net=self.net,
    experiment_name=self.experiment_name,
    ckpt_dir=self.checkpoints_dir_path,
    criterion=self.criterion,
    lr_warmup_epochs=self.training_params.lr_warmup_epochs,
    sg_logger=self.sg_logger,
    train_loader=self.train_loader,
    valid_loader=self.valid_loader,
    training_params=self.training_params,
    ddp_silent_mode=self.ddp_silent_mode,
    checkpoint_params=self.checkpoint_params,
    architecture=self.architecture,
    arch_params=self.arch_params,
    metric_to_watch=self.metric_to_watch,
    device=device_config.device,
    ema_model=self.ema_model,
)
```
However, in the `test` function, the context is initialized with fewer parameters (`criterion`, `device`, `sg_logger`). Here is the related code:
```python
context = PhaseContext(
    criterion=self.criterion,
    device=self.device,
    sg_logger=self.sg_logger,
)
```
### Suggested Enhancement
It would be beneficial to extend the context parameters in the `test` function to match those in the `train` function. This will facilitate the use of callback functions that may require additional context information, such as `net`.

**Comment by BloodAxe** [comment]
Yes, that would be nice improvement. Are you interested in contributing this change?
